src/main/java/com/fasterxml/jackson/core/util/BufferRecyclerProvider.java

```diff
@@ -0,0 +1,19 @@
+package com.fasterxml.jackson.core.util;
+
+import com.fasterxml.jackson.core.TokenStreamFactory;
+
+/**
+ * Provider for {@link BufferRecycler}s to allow pluggable and optional
+ * recycling of underlying input/output buffers.
+ *
+ * @since 2.16
+ */
+public abstract class BufferRecyclerProvider
+    implements java.io.Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    public abstract BufferRecycler acquireBufferRecycler(TokenStreamFactory forFactory);
+
+    public abstract void releaseBufferRecycler(BufferRecycler r);
```

Comment:
Not 100% sure this is needed by life-cycle, but I think it may be.

Reply:
In the most general case I believe that this method is necessary, even though for some implementations, like the `ThreadLocal` based one, it could be empty. If you turn this abstract class into an interface as I was suggesting, maybe you can also provide a `default` implementation for this method with an empty body?